packages/sign/src/witness/tlog/index.ts

```diff
@@ -32,15 +34,26 @@ type TransparencyLogEntries = { tlogEntries: TransparencyLogEntry[] };
 
 export type RekorWitnessOptions = Partial<TLogClientOptions> & {
   entryType?: 'dsse' | 'intoto';
+  majorApiVersion?: number;
 };
 
 export class RekorWitness implements Witness {
-  private tlog: TLog;
+  private tlogV1: TLog;
+  private tlogV2: TLogV2;
   private entryType?: 'dsse' | 'intoto';
+  private majorApiVersion: number;
 
   constructor(options: RekorWitnessOptions) {
     this.entryType = options.entryType;
-    this.tlog = new TLogClient({
+    this.majorApiVersion = options.majorApiVersion || 1;
+
+    this.tlogV1 = new TLogClient({
+      ...options,
+      rekorBaseURL:
+        options.rekorBaseURL || /* istanbul ignore next */ DEFAULT_REKOR_URL,
```

Comment:
I believe we would have a following up PR for handling the rekorBaseURL and the majorApiVersion?